plugins/carbon_black_cloud/icon_carbon_black_cloud/connection/connection.py

```diff
@@ -146,3 +152,78 @@ def test(self):
                 data=str(error),
             )
         return {"success": True}
+
+    def test_task(self):
+        self.logger.info("Running a connection test to Carbon Black Cloud")
+
+        alerts_endpoint = self.base_url + f"/api/alerts/v7/orgs/{self.org_key}/alerts/_search"
+        observations_endpoint = self.base_url + f"/api/investigate/v2/orgs/{self.org_key}/observations/search_jobs"
+
+        current_time = datetime.now(timezone.utc)
+        now = current_time.strftime(TIME_FORMAT)
+        minus_five_mins = (current_time - timedelta(minutes=5)).strftime(TIME_FORMAT)
+
+        fd = {
+            alerts_endpoint: {
+                "name": "alerts",
+                "params": {
+                    "time_range": {"start": minus_five_mins, "end": now},
+                    "criteria": {},
+                    "start": "1",
+                    "rows": str(1),
+                    "sort": [{"field": ALERT_TIME_FIELD, "order": "ASC"}],
+                },
+            },
+            observations_endpoint: {
+                "name": "observations",
+                "params": {
+                    "rows": 1,
+                    "start": 0,
+                    "fields": ["*"],
+                    "criteria": {"observation_type": OBSERVATION_TYPES},
+                    "sort": [{"field": OBSERVATION_TIME_FIELD, "order": "asc"}],
+                    "time_range": {"start": minus_five_mins, "end": now},
+                },
+            },
+        }
+
+        return_msg = "Task connection test to Carbon Black Cloud successful"
+        return_msg_error = "Task connection test to Carbon Black Cloud failed"
+
+        failed = ""
+        failed_endpoint = []
+
+        for key, value in fd.items():
+            self.logger.info(f"Testing {value.get('name')} endpoint")
+
+            try:
+                self.request_api(key, value.get("params"))
+
+            except HTTPErrorException as error:
+                # If we get a similar error for the second URL
+                if error.cause in failed:
+                    failed_endpoint.append(value)
+                else:
+                    return_msg_error += f"\n{error.cause}\n{error.assistance}"
+                    failed_endpoint.append(value)
+                    failed += return_msg_error
+
+            except RateLimitException as error:
+                return_msg_error += "\nToo many requests."
+                return_msg_error += "\nPlease slow down"
+                raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=return_msg_error)
+
+            except PluginException as error:
+                return_msg_error += "\nRetry on 503 failed."
```

Comment:
Think we should change this as well to say 'Carbon Black currently unavailable. Please try again'. Just to remove the 503 from the user facing message. Better to log this and the RatelimitException for our debugging?